### PR TITLE
openxray: 784-october-preview -> 822-december-preview

### LINKED ISSUE #119299: OpenXRay: 822-december-preview available
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: openxray

current version: 784-october-preview
desired version: 822-december-preview

###### Notify maintainers

maintainers: @gnidorah (Account was deleted)

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Motivation for this change
Closes #119299

Done:
- Bump
- Fixing pname not matching name in all-packages.nix
- Fixing format hardening
- Removing @gnidorah (account deleted) and adding myself as maintainer
- Tested with game files

Bildschirmfoto von 2021-05-05 23-14-44

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fufexan** [comment]
Result of `nixpkgs-review pr 121806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>openxray</li>
  </ul>
</details>

**Review by fufexan — APPROVED** [review]

File: pkgs/games/openxray/default.nix [hunk 1]
```diff
@@ -35,25 +35,29 @@ let
...
-  pname = "OpenXRay";
+  pname = "openxray";
   inherit version src;
 
-  hardeningDisable = [ "format" ];
-  cmakeFlags = [ "-DCMAKE_INCLUDE_PATH=${cryptopp}/include/cryptopp" ];
-
+  # Fixes hardening
   patches = [
     (fetchpatch {
```
Comment (jonringer) [inline pkgs/games/openxray/default.nix:43]: do you mind adding a comment about what the intent of the patch is doing, and when the patch is likely not to be needed
```
    # does x, y, z. Should be able to be removed after >=X.Y.Z
    (fetchpatch {
```

Comment (OPNA2608) [reply pkgs/games/openxray/default.nix:44]: Is this better? I don't know when that PR gets merged and upstream's next version name isn't as predictable as x.y.z.

Bildschirmfoto von 2021-05-07 09-14-47

**Comment by OPNA2608** [comment]
Fixed merge conflict.

File: pkgs/games/openxray/default.nix [hunk 2]
```diff
@@ -35,25 +35,30 @@ let
...
-  hardeningDisable = [ "format" ];
-  cmakeFlags = [ "-DCMAKE_INCLUDE_PATH=${cryptopp}/include/cryptopp" ];
-
+  # TODO https://github.com/OpenXRay/GameSpy/pull/6, check if merged in version > 822
+  # Fixes format hardening
   patches = [
     (fetchpatch {
-      url = "https://github.com/OpenXRay/xray-16/commit/4532cba11e98808c92e56e246188863261ef9201.patch";
-      sha256 = "1hrm4rkkg946ai95krzpf3isryzbb2vips63gxf481plv4vlcfc9";
+      url = "https://github.com/OpenXRay/GameSpy/pull/6/commits/155af876281f5d94f0142886693314d97deb2d4c.patch";
```
Comment (SuperSandro2000) [inline pkgs/games/openxray/default.nix:45]:
```suggestion
      url = "https://github.com/OpenXRay/GameSpy/commit/155af876281f5d94f0142886693314d97deb2d4c.patch";
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 121806 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>openxray</li>  </ul></details>

File: pkgs/games/openxray/default.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/games/openxray/default.nix:45]: I don't like that format, it requires an extra comment with a link to the PR to properly track the merge status. The ".patch"-less commit link just tells you "this commit doesn't belong to any branch on this repository".

**Pull request merged**